webapp/src/js/components/panoptes/MuiDataTableView.js

```diff
@@ -273,64 +310,97 @@ let MuiDataTableView = createReactClass({
               </TableRow>
             </TableHead>
             <TableBody>
-              {data.map((row, rowIndex) =>
-                <TableRow
-                  hover
-                  key={'row_' + rowIndex}
-                  style={{cursor: 'pointer'}}
-                  onClick={(e) => this.handleClick(e, row[primaryKeyColumnId])}
-                >
-                  {groupOrderedColumns.map((column, columnIndex) => {
-                    const columnData = this.propertiesByColumn(column);
-                    let {maxVal, minVal, alignment, valueColours, showBar} = columnData;
-                    let cellData = row[columnData.id];
-
-                    let background = 'inherit';
-                    if (showBar && cellData !== null && maxVal !== undefined && minVal !== undefined) {
-                      cellData = parseFloat(cellData);
-                      let percent = 100 * (cellData - minVal) / (maxVal - minVal);
-                      background = `linear-gradient(to right, ${rowIndex % 2 ? 'rgb(115, 190, 252)' : 'rgb(150, 207, 253)'} ${percent}%, rgba(0,0,0,0) ${percent}%`;
-                    } else if (cellData !== null && maxVal !== undefined && minVal !== undefined) {
-                      let clippedCellData = Math.min(Math.max(parseFloat(cellData), minVal), maxVal);
-                      background = Color(MAX_COLOR).lighten(0.3 * (1 - (clippedCellData - minVal) / (maxVal - minVal))).string();
-                    }
-                    if (valueColours) {
-                      let col = valueColours[cellData] || valueColours['_other_'];
-                      if (col) {
-                        col = Color(col).lighten(0.3);
-                        if (rowIndex % 2)
-                          col.darken(0.1);
-
-                        background = col.string();
+              {data.map((row, rowIndex) => {
+                const primKey = row[primaryKeyColumnId];
+
+                let onClickComponentPropsJSON = onClickComponentProps !== undefined ? JSON.parse(onClickComponentProps) : {};
```

Comment:
Why pass the component props as a string? I think if it is defined as `object` in prop types then the template system will parse a JSON string.

Reply:
Ah yes, I forgot about that (again?), thanks. (I didn't need to use `onClickComponentProps` in any of the implementations, since `table` and `primKey` are included and passed by default), so I only included this support for completeness. YAGNI schmagni!) Will amend, test and PR.